### DISCUSSION TITLE
Why is Universal CRT required?

### DISCUSSION THREAD

**Discussion by jay** [curl/curl-for-win#43] [Q&A] [answered]
I can see that last year CRT dependency was switched to from msvcrt to Universal, why? I'm using an old Windows 7 VM snapshot and I had to [download](https://support.microsoft.com/en-us/topic/update-for-universal-c-runtime-in-windows-c0514201-7fe6-95a3-b0a5-287930f3560c) the Universal CRT to it when I saw a cryptic error about curl not starting due to a missing dll. Not a problem but I wasn't expecting that.

Edit: I did not have this problem in a more recent Windows 7 VM snapshot. The update was already included, presumably by Windows Update.

**Comment by vszakats** [comment, marked as answer]
- Legacy `msvcrt` is a piece of code that hasn't seen an update or bugfix in many years. It's abandoned/unmaintained code. It also had accumulated unfixed bugs/vulnerabilities. Though even without the latter, relying on abandoned components I feel is best to avoid.
According to MS [^1] [^2], on Windows 7 UCRT was supposed to get installed via Windows Update. I haven't double-checked their claims, and cannot tell why it wasn't present on your system. Either way it can be installed via an official update package (also for Vista).

- UCRT is required for ARM64 builds.
- Using UCRT for ARM64 while keeping `msvcrt` for Intel is overly complex and error-prone.

[^1]: https://learn.microsoft.com/en-us/cpp/porting/upgrade-your-code-to-the-universal-crt?view=msvc-170#deployment-and-redistribution-of-the-universal-crt

[^2]: https://learn.microsoft.com/en-us/cpp/windows/universal-crt-deployment?view=msvc-170
